controller/src/main/java/io/pravega/controller/store/stream/Int96CounterImpl.java

```diff
@@ -95,7 +100,7 @@ CompletableFuture<Void> refreshRangeIfNeeded() {
 
                     // Need to refresh counter and limit. Start a new refresh future. We are under lock so no other
                     // concurrent thread can start the refresh future.
-                    refreshFutureRef = getRefreshFuture()
+                    refreshFutureRef = function.apply(new CounterInfo(context, COUNTER_RANGE), this::reset)
```

Comment:
same here

Reply:
We need to have a lock as **refreshFutureRef** is member variable and we don't want to concurrently update this member. Having a lock on **refreshFutureRef** assignment will help us to protect it from concurrent update.

Reply:
Yes. This is why I suggested SequentialAsyncProcessor

Reply:
I have used SequentialProcessor.